### PR TITLE
Reduce Missing Joint Log Spam

### PR DESCRIPTION
**Pull request by kasenvr**
This PR reduces the log spam for a missing joint sent by the AnimInverseKinematics to 30 second intervals instead of many dozens of times a second. To test, use an avatar that normally gives you this log spam and notice how the message will only show once every 30 seconds.

### CODE REVIEW

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-69b8730_aaf19db2.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-69b8730.tar_aaf19db2.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-69b8730_aaf19db2.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-69b8730.tar_aaf19db2.gz

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-69b8730_aaf19db2.exe

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-eadfb40.tar_2faef4f0.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-eadfb40_2faef4f0.sh
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-eadfb40.tar_2faef4f0.gz

**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-eadfb40_2faef4f0.dmg

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-eadfb40_2faef4f0.exe

File: libraries/animation/src/AnimInverseKinematics.cpp [hunk 1]
```diff
@@ -158,6 +159,16 @@ void AnimInverseKinematics::setTargetVars(const QString& jointName, const QStrin
     }
 }
 
+bool debounceJointWarnings() {
+    static QTime last_call;
```
Comment (ctrlaltdavid) [inline libraries/animation/src/AnimInverseKinematics.cpp:163]: Would probably be better to declare this QTime near the start of the file, e.g., directly after `JOINT_WARNING_DEBOUNCE_TIME`.
Also, name the object, say, `debounceJointWarningsClock` - variable name with "_" is non-standard.

File: libraries/animation/src/AnimInverseKinematics.cpp [hunk 2]
```diff
@@ -25,6 +25,7 @@
 
 static const int MAX_TARGET_MARKERS = 30;
 static const float JOINT_CHAIN_INTERP_TIME = 0.5f;
+static const int WARNING_DEBOUNCE_TIME = 30000; // 30 seconds
```
Comment (ctrlaltdavid) [inline libraries/animation/src/AnimInverseKinematics.cpp:28]:
```suggestion
static const int JOINT_WARNING_DEBOUNCE_TIME = 30000; // 30 seconds
```

File: libraries/animation/src/AnimInverseKinematics.cpp [hunk 3]
```diff
@@ -158,6 +159,16 @@ void AnimInverseKinematics::setTargetVars(const QString& jointName, const QStrin
     }
 }
 
+bool debounceJointWarnings() {
+    static QTime last_call;
+
+    if (last_call.elapsed() >= WARNING_DEBOUNCE_TIME || !last_call.isValid()) {
```
Comment (ctrlaltdavid) [inline libraries/animation/src/AnimInverseKinematics.cpp:165]: Should need to test `!last_call.isValid()` if you include the following code in the `AnimInverseKinematics::AnimInverseKinematics()` constructor:
```
debounceJointWarningsClock.start();
```

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-3627584_4c780a59.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-3627584.tar_4c780a59.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-3627584.tar_4c780a59.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-3627584_4c780a59.sh

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-2f06556_0ad7e25b.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-2f06556.tar_0ad7e25b.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-2f06556.tar_0ad7e25b.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-2f06556_0ad7e25b.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-2f06556_0ad7e25b.exe

**Comment by vircadia-build-notifier** [comment]
The following links are available: 
**build (macOS-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-837805c_44e5bbb5.dmg

**build (ubuntu-18.04, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-837805c.tar_44e5bbb5.gz
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-837805c.tar_44e5bbb5.Z
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-837805c_44e5bbb5.sh

**build (windows-latest, full)**
 - https://athena-public.s3.eu-west-3.amazonaws.com/builds/PR/kasenvr_project-athena/PR474/Vircadia-Alpha-PR474-837805c_44e5bbb5.exe

**Comment by SilverfishVR** [comment]
👍 looks to be working as intended
Screenshot 2020-07-04 16 10 53

**Pull request merged**